Source/Fuse.Controls.DatePicker/iOS/DatePicker.uno

```diff
@@ -0,0 +1,188 @@
+using Uno;
+using Uno.UX;
+using Uno.Compiler.ExportTargetInterop;
+
+using Fuse;
+using Fuse.Controls;
+using Fuse.Controls.Native;
+using Fuse.Controls.Native.iOS;
+
+namespace Fuse.Controls.Native.iOS
+{
+	extern(iOS) static class DateTimeConverterHelpers
+	{
+		const long DotNetTicksInSecond = 10000000L;
+		const long UnixEpochInDotNetTicks = 621355968000000000L;
+
+		public static DateTime ConvertNSDateToDateTime(ObjC.Object date)
+		{
+			var secondsSince1970InUtc = (long)NSDateToSecondsSince1970InUtc(date);
+
+			var dotNetTicksRelativeToUnixEpoch = (long)secondsSince1970InUtc * DotNetTicksInSecond;
+			var dotNetTicks = dotNetTicksRelativeToUnixEpoch + UnixEpochInDotNetTicks;
+
+			return new DateTime(dotNetTicks, DateTimeKind.Utc);
+		}
+
+		public static ObjC.Object ConvertDateTimeToNSDate(DateTime dt)
+		{
+			dt = dt.ToUniversalTime();
+
+			var dotNetTicks = dt.Ticks;
+			var dotNetTicksRelativeToUnixEpoch = dotNetTicks - UnixEpochInDotNetTicks;
+			var secondsSince1970InUtc = dotNetTicksRelativeToUnixEpoch / DotNetTicksInSecond;
+
+			return SecondsSince1970InUtcToNSDate((double)secondsSince1970InUtc);
+		}
+
+		[Foreign(Language.ObjC)]
+		public static double NSDateToSecondsSince1970InUtc(ObjC.Object date)
+		@{
+			return [date timeIntervalSince1970];
+		@}
+
+		[Foreign(Language.ObjC)]
+		public static ObjC.Object SecondsSince1970InUtcToNSDate(double secondsSince1970InUtc)
+		@{
+			return [NSDate dateWithTimeIntervalSince1970:secondsSince1970InUtc];
+		@}
+
+		[Foreign(Language.ObjC)]
+		public static ObjC.Object ReconstructUtcDate(ObjC.Object date)
+		@{
+			if (!date)
+				return [NSDate dateWithTimeIntervalSince1970:0];
+
+			// Reconstruct the same date in UTC without time components
+			NSCalendar *utcCalendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierGregorian];
+			[utcCalendar setTimeZone:[NSTimeZone timeZoneForSecondsFromGMT:0]];
+
+			NSDateComponents *components = [utcCalendar components:NSYearCalendarUnit|NSMonthCalendarUnit|NSDayCalendarUnit fromDate:date];
+
+			NSDateComponents *utcComponents = [[NSDateComponents alloc] init];
+			[utcComponents setYear:[components year]];
+			[utcComponents setMonth:[components month]];
+			[utcComponents setDay:[components day]];
+
+			return [utcCalendar dateFromComponents:utcComponents];
+		@}
+	}
+
+	extern(!iOS) class DatePickerView
+	{
+		[UXConstructor]
+		public DatePickerView([UXParameter("Host")]IDatePickerHost host) { }
+	}
```

Comment:
Do you really need this? Having two classes with the same name is kind of nasty for the doc-generator etc. These classes doesn't have docs, so it probably isn't an issue for that reason, but it seems a bit pointless...

Reply:
Good call, I think I added that when testing only one platform impl and forgot to remove it. I'll see if it's needed and remove it if not.

Reply:
aha, now I remember why I have that. Otherwise the UX compiler complains about missing types in `DatePicker.ux`:

```ux
<Fuse.Controls.DatePickerBase ux:Class="Fuse.Controls.DatePicker" ux:Name="self">
	<Panel ux:Template="GraphicsAppearance" Margin="4" Background="#eee" Padding="30">
		<Text Value="GraphicsAppearance for DatePicker not implemented!" Alignment="Center" TextWrapping="Wrap" />
		<Rectangle Layer="Background">
			<Stroke Color="#000" />
		</Rectangle>
	</Panel>
	<Fuse.Controls.Native.iOS.DatePickerView ux:Condition="iOS" ux:Template="iOSAppearance" Host="self" />
	<Fuse.Controls.Native.Android.DatePickerView ux:Condition="Android" ux:Template="AndroidAppearance" Host="self" />
</Fuse.Controls.DatePickerBase>
```

I would've assumed `ux:Condition` would have solved that, but perhaps that's only evaluated at runtime.

This pattern comes from [our current docs](https://www.fusetools.com/docs/native-interop/native-ux-components) btw.

Reply:
OK, that's a sucky implementation of the UX compiler in this regard, but OK. It is what it is ;)

Reply:
agreed, https://github.com/fusetools/uno/issues/1348